ReactWindows/ReactNative/Accessibility/AccessibilityHelper.cs

```diff
@@ -218,8 +223,8 @@ public static void OnChildAdded(UIElement parent, DependencyObject child)
             // No guarantee all children are hidden anymore due to additional child, reset the flag
             SetCurrentlyHidingChildren(parent, ElementAccessibilityContext.HidingChildren.NotSure);
 
-            // Mark the parent dirty
-            MarkElementDirty(parent);
+            // Mark the parent dirty (or skip if there is no associated peer available)
+            MarkElementDirty(parent, true);
```

Comment:
I think that the parent element may have no peer but the the peer chain may still exists. So my take on this is that the parent peer should be accessed via the child peer i.e. obtain the peer from the child that is being added that then use the AutomationPeer.Navigate(Parent) to obtain the automationpeer for the parent. Since the parent in the automation tree can be few levels above the "parent" that we have in the call, i.e automation tree may miss nodes that exist in the RNW parent/child tree but these will be irrelevant to automation.

Reply:
Yes, this doesn't seem to fix the thing completely. It worked well in Narrator, but that may be due to "Eventual convergence".. The added nodes are always marked as "don't know if hiding children", so some traversal may find them and process them all, eventually.
But I agree this is not proof of correctness.
I'll have to think more about it. Your solution is right in some cases, but there are others that are still not covered (let's say the added child is has no peer either... it's an already added grandchild that has a peer, and that element is already marked dirty, but the peer had no parent when that happened. I would have to get to that peer, but how? I suspect that the peer has a parent now, some ancestor of our "parent" element, but I'm somewhere in the middle, scratching my head on how to make this deterministic AND performant...




---
In reply to: [178356157](https://github.com/Microsoft/react-native-windows/pull/1748#discussion_r178356157) [](ancestors = 178356157)